sat/config.py

```diff
@@ -342,13 +343,18 @@ def get_config_value(query_string):
     if len(parts) != EXPECTED_LEVELS:
         raise ValueError("Wrong number of levels in query string passed to get_config_value(). "
                          "(Should be {}, was {}.)".format(EXPECTED_LEVELS, len(parts)))
-    else:
-        section, option = parts
-        if not section or not option:
-            raise ValueError("Improperly formatted query string supplied to get_config_value(). "
-                             "(Got '{}'.)".format(query_string))
-        else:
-            return CONFIG.get(section, option)
+
+    section, option = parts
+    if not section or not option:
+        raise ValueError("Improperly formatted query string supplied to get_config_value(). "
+                         "(Got '{}'.)".format(query_string))
+
+    value = CONFIG.get(section, option)
+
+    if option.lower() == 'timeout' and value == '-1':
+        return math.inf
```

Comment:
It looks to me like this change is not actually being used, so I'd suggest removing it. You're handling a timeout value of -1 in the `do_parallel_bos_operations` function instead, and I think that's fine, but this code will be confusing if left here.

Reply:
I opened CRAYSAT-1916 for this.